docs/user/commit-message.md

```diff
@@ -0,0 +1,112 @@
+# Format and Update Commit Messages
+
+You should format your commit messages like this:
+
+```
+One-line title
+
+Then a description, which may be multiple lines long.
+This describes the change you are making with this commit.
+
+Test Plan: how to test the change in this commit.
+
+The test plan can also be several lines long.
+
+Reviewers: github-username-a, github-username-b
+```
+
+The first line will be the title of the PR created by `spr diff`, and the rest of the lines except for the `Reviewers` line will be the PR description (i.e. the content of the first comment). The GitHub users named on the `Reviewers` line will be added to the PR as reviewers.
+
+The `Test Plan` section is required to be present; `spr diff` will fail with an error if it isn't.
+
+## Updating the commit message
+
+When you create a PR with `spr diff`, **the PR becomes the source of truth** for the title and description. If you later amend the commit message locally, subsequent runs of `spr diff` will fail with an error. Additionally, when you land a commit with `spr land`, its commit message will be amended to match the PR's title and description, regardless of what is in your local repo.
+
+If you want to update the title or description, there are two ways to do so:
+
+- Modify the PR through GitHub's UI.
+
+- Amend the commit message locally, then run `spr diff --update-message`. _Note that this does not update reviewers_; that must be done in the GitHub UI.
+
+If you want to go the other way --- that is, make your local commit message match the PR's title and description --- you can run `spr amend`.
+
```

Comment:
It sounds so negative to say spr will fail with an error without immediately following up how to fix this. Maybe this can be moved to the end.

```suggestion
When you create a PR with `spr diff`, **the PR becomes the source of truth** for the title and description. When you land a commit with `spr land`, its commit message will be amended to match the PR's title and description, regardless of what is in your local repo.

If you want to update the title or description, there are two ways to do so:

- Modify the PR through GitHub's UI.

- Amend the commit message locally, then run `spr diff --update-message`. _Note that this does not update reviewers_; that must be done in the GitHub UI.

If you want to go the other way --- that is, make your local commit message match the PR's title and description --- you can run `spr amend`.

When you call `spr diff` to update an existing PR, it will fail with an error if the local commit message does not match the PR title and description. In that case, proceed with either `spr diff --update-message` if you want to use your local commit message (the PR will be updated accordingly) or `spr amend` if you want to use what is on GitHub (and this will update the message of your local commit).
```

Reply:
Oh man, I wish GitHub was better at presenting suggested changes. I only modified the first paragraph and added one at the end...